### PR TITLE
Be more precise in deciding whether to add the schema prelude

### LINKED ISSUE #175: Invalid schema on schema.graphql
invalid schema: Undefined type String.
 
 for every type, I am getting the same error
 
```
 type ABC implements D & E @key(fields: "id"){
    style: Style
    """ ID of the Style """
    id: ID!
    """ Style name """
    stylename: String
}
```

### PR DESCRIPTION
**Pull request by benjaminjkraft**
GraphQL schemas have some builtin types, like `String`. The spec says
your SDL must not include those, but in practice some schemas do. (This
is probably because introspection must include them, and some tools that
create SDL from introspection don't know they're supposed to filter them
out.) Anyway, we've since #145 had logic to handle this; we just parse
with and without the prelude that defines them and see which works.

The problem is that this makes for very confusing error messages if you
have an invalid schema. (Or if you have a schema that you think is valid
but gqlparser doesn't, which is the more common case in the wild; see
for example #200.) Right now if both ways error we take the
without-prelude error, which if you didn't define the builtins is just
`undefined type String`; if we took the with-prelude error then if you
did define the builtins you'd just get `type String defined twice`. So
we actually have to be smart if we want good error messages for
everyone.

So in this commit we are smart: we check if your schema defines
`String`, and include the prelude only if it does not. To do this I
basically inlined `gqlparser.LoadSchema` (twice), so that in between
parsing and validation we can check if you have `String` and if not add
the prelude. This should in theory be both more efficient
(we don't have to do everything twice) and give better error messages,
although it's a bit more code.

Fixes #175.

Test plan: make check

I have:
- [x] Written a clear PR title and description (above)
- [x] Signed the [Khan Academy CLA](https://www.khanacademy.org/r/cla)
- [x] Added tests covering my changes, if applicable
- [x] Included a link to the issue fixed, if applicable
- [x] Included documentation, for new features
- [x] Added an entry to the changelog

### CODE REVIEW

**Review by csilvers — APPROVED** [review]
It may be a bit more code, but I think it's clearer!  Nice.
